gateway-messages/src/sp_to_mgs.rs

```diff
@@ -1033,3 +1040,27 @@ impl fmt::Display for VpdError {
         }
     }
 }
+
+/// Watchdog errors encountered configuring the SP-RoT watchdog
+///
+/// This value is wrapped by [`SpError`]
+#[derive(
+    Debug, Clone, Copy, Eq, PartialEq, SerializedSize, Serialize, Deserialize,
+)]
+pub enum WatchdogError {
+    /// Could not control the SP over SWD
+    SpCtrl,
+    /// There is not a complete SP update in place
+    NoCompletedUpdate,
+}
+
+impl fmt::Display for WatchdogError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::NoCompletedUpdate => {
```

Comment:
Oh, this might answer my earlier question about the potential TOCTOU (I clearly haven't looked at the hubris side of this yet!). Is this right?

* If we try to reset with watchdog to an SP that does not have a completed update, it will return this error?
* If so, (a) the check for a completed update _is_ required and (b) the TOCTOU is solved by this on the SP side (since if one client checks for an update, then gets paused while another client resets the SP, then resumes, the reset will fail with this error)?

Reply:
Yes, your interpretation is correct.  There shouldn't be a TOCTOU issue because the SP will reject the `ResetComponentTriggerWithWatchdog` command if there's no update staged, so the check earlier is just for more ergonomic error handling.